docs/node2node.md

```diff
@@ -0,0 +1,105 @@
+# node to node messages
+
+
+[summary placeholder]
+
+(todo: fully describe pop and shift mechanism)
+(todo: calcuate optimal data sizes)
+
+
+## streamed requests
+
+direct node to node request/response messages sent on a bidi quic stream.
+
+the stream starts by sending 2 bytes identifying the stream kind of which there are the following:
+
+* network data lookup
+* network data upsert
+* path build
+* path update
+* path stream
```

Comment:
the idea is that streams are for one type of request, but there is no
requirement that you only have one per type. this is especially true
with path stream type.

looking at this again i think it would be best to define which streams
would allow multiple instances per remote. e.g. a path stream
connection would have many instances to each remote, and per path
per remote, but for network data lookup/upsert you'd really only need
one per remote.